ydb/docs/en/core/contributor/datashard-distributed-txs.md

```diff
@@ -0,0 +1,158 @@
+5. As long as the transaction is in the queue, and especially while its `PlanStep` is not yet known, incoming ReadSets are stored in memory, and will not be acknowledged. Acknowledgements are delayed until the transaction has been fully committed or is aborted prematurely (in which case, no changes have been made and no transaction record has been committed, so newer generations cannot recover or commit the transaction).
+
+Volatile distributed transactions also ensure that changes are visible and stable across multiple shards participating in the same distributed transaction. For instance, persistent distributed transactions allow the following anomaly:
+
+1. Distributed transaction `Tx1` performs a blind write to keys `x` and `y` in two different shards.
+2. Single-shard transaction `Tx2` reads key `x` and observes a value written by `Tx1`.
+3. Single-shard transaction `Tx3` (which starts after `Tx2` has completed) reads key `y` and does not observe a value written by `Tx1`.
+
+This anomaly can occur when a shard with key `x` quickly receives a transaction plan step with `Tx1` and performs a write. A subsequent read in `Tx2`, which arrives a bit later, will have its local MVCC version include changes to key `x` and observe the change. However, the shard with key `y` may be running on a slower node and may not be aware of `Tx1` plan step when a subsequent read in `Tx3` arrives. Since the time cast bucket doesn't have `Tx1` acknowledged, its mediator time will remain in the past and `Tx3` will have a local MVCC version that doesn't include changes to key `y`. From the user's perspective, `Tx2` must happen before `Tx3`, but the global serializable order turns out to be `Tx3`, `Tx1`, `Tx2`.
+
+Volatile transactions cannot observe this particular anomaly because a read that observes changes to key `x` must have received `DECISION_COMMIT` from the other shard with key `y`. This means that the transaction record has been persisted in both shards and, crucially, the shard with key `y` has marked it as completed. A read from key `y` will choose a local MVCC version that includes changes made by `Tx1` and will need to wait until `Tx1` is resolved.
+
+In other words, if a client observes the result of a volatile transaction at a certain point in time, all subsequent reads will also observe the result of that volatile transaction, and changes stability is not violated.
+
+### Indirect planning of volatile transactions
+
+The coordinator restart may cause the plan steps to only reach a subset of the participants in a volatile distributed transaction. As a result, some shards may execute the transaction and begin waiting for ReadSets, while other shards continue to wait for the plan step to arrive. Due to the planning timeout of 30 seconds, this could cause some transactions to experience excessive delays before eventually aborting.
```

Comment:
```suggestion
The coordinator restart may cause the plan step to only reach a subset of the participants in a volatile distributed transaction. As a result, some shards may execute the transaction and begin waiting for ReadSets, while other shards continue to wait for the plan step to arrive. Due to the planning timeout of 30 seconds, this could cause some transactions to experience excessive delays before eventually aborting.
```